### ISSUE TITLE
[SERVE] Allow adjustment of scaling policies without redeployment

### ISSUE THREAD

**Issue by JGSweets** [skypilot-org/skypilot#4442]
Currently, when altering the `replica_policy`, update runs a pseudo blue-green deployment in the sense it launches all new resources.

Preferably, if only the replica_policy is changing, it alters the policy itself without deploying /tearing down new instances unless required by the new policy.

---
Example 1:
Init:      Currently, 2 resources are running, but the `min_replica` is set to 3.
Result: Only 1 instance is launched.

Example 2:
Init:      Currently, 2 resources are running, but the `min_replica` is set to 1 and qps would not be met if scaled down.
Result: 1 instance is torn down.

---
Solution Options:
1. Have update check to make sure only the replica_info has changed with a hash.
2. Use a flag with update that allows altering just the replica policy
3. Have a separate update endpoint which allows updating the replica policy.

---
_Version & Commit info:_
skypilot, version 0.7.0
skypilot, commit 3f62588

**Comment by Michaelvll** [comment]
cc'ing @cblmemo

**Comment by cblmemo** [comment]
Hi @JGSweets , thanks for reporting this! However, I think we already have this feature, see:

https://github.com/skypilot-org/skypilot/blob/ee3cabd57247ff0f25cb65c0ee46bd35ead8d11a/sky/serve/replica_managers.py#L1189-L1223

I also tried the following on current master (`ee3cabd57247ff0f25cb65c0ee46bd35ead8d11a`):

```bash
# now `service.replicas` field is 1
sky serve up -n minimal examples/serve/minimal.yaml
# change `service.replicas` field to 2
sky serve update minimal examples/serve/minimal.yaml
```

and got the following:

<img width="856" alt="image" src="https://github.com/user-attachments/assets/f9dfc025-cc48-417d-915f-2a3a06f0acea" />

Noticed that the replica with ID 1 has a version of 2, which means this replica is reused.

Could you share more of your usage? If that does not works for you, it is possible that there are some bugs in our system, and some related tests could help us find it ;)

**Comment by JGSweets** [comment]
Interesting, I had an experience recently where I increased `min_replicas` 2->4 and `max_replicas` 3->4, but it scaled up 4 new resources like the blue green deployment mentioned in the docs.

**Comment by JGSweets** [comment]
Is the intended functionality mentioned in the docs?

**Comment by cblmemo** [comment]
> Is the intended functionality mentioned in the docs?

Yes, pls check the first hint in this doc: https://docs.skypilot.co/en/latest/serving/update.html

**Comment by JGSweets** [comment]
I'll have to see if I can tease the issue out more / can replicate it. I thought using old resources was the intended functionality, so glad you confirmed it.


I'd need to verify that ami didn't change ain addition to the min/max replicas. Possibly what happened, but I thought it was the same.
